libs/http-utils/src/response/api-gateway-response.ts

```diff
@@ -0,0 +1,103 @@
+import type { APIGatewayProxyResult } from 'aws-lambda';
+import type { ResponseBodyConfig } from './shared/response-types';
+import {
+  processResponseBody,
+  serializeBody,
+} from './shared/response-body-processor';
+import { mergeHeaders } from './shared/response-headers';
+import type { ResponseOptions } from './shared/response-options';
+
+/**
+ * Creates a generic API Gateway Proxy Result response
+ * @param options - Response configuration options
+ * @param body - Response body (will be stringified if not already a string)
+ * @returns APIGatewayProxyResult
+ */
+function createApiGatewayResponse(
+  options: ResponseOptions,
+  body?: unknown,
+): APIGatewayProxyResult {
+  const response: APIGatewayProxyResult = {
+    statusCode: options.statusCode,
+    headers: mergeHeaders(options.headers),
+    body: serializeBody(body),
```

Comment:
Do we want to enforce a consistent shape to the error response body?

i.e.

```
description: 'foo',
error: 'foo'
```

```
createErrorResponse()
```

(Note the Shared Signals error's should be of an explicit [shape](https://govukverify.atlassian.net/browse/GOVUKAPP-2404)).

So we could extend the error response to:

```
createSETErrorResponse()
```

Reply:
Feels like we'd be creating specific business logic responses here 🤔